packages/wonder-blocks-dropdown/components/dropdown-opener.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+
+import {ClickableBehavior} from "@khanacademy/wonder-blocks-core";
+
+import type {
+    AriaProps,
+    ClickableHandlers,
+    ClickableState,
+} from "@khanacademy/wonder-blocks-core";
+
+type Props = {|
+    ...$Rest<AriaProps, {|"aria-disabled": "true" | "false" | void|}>,
+
+    /**
+     * The child function that returns the anchor the Dropdown will be activated
+     * by. This function takes `eventState`, which allows the opener element to
+     * access pointer event state.
+     */
+    children: (eventState: ClickableState) => React.Element<any>,
+
+    /**
+     * Whether the opener is disabled. If disabled, disallows interaction.
+     */
+    disabled: boolean,
+
+    /**
+     * Callback for when the opener is pressed.
+     */
+    onClick: (e: SyntheticEvent<>) => mixed,
+
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string,
+
+    /**
+     * Callback used to pass the childs Ref back up to the parent element
+     */
+    anchorRef: (?HTMLElement) => mixed,
+|};
+
+class DropdownOpener extends React.Component<Props> {
+    static defaultProps = {
+        disabled: false,
+    };
+
+    componentDidMount() {
+        const anchorNode = ((ReactDOM.findDOMNode(this): any): ?HTMLElement);
+        this.props.anchorRef(anchorNode);
+    }
+
+    renderAnchorChildren(
+        eventState: ClickableState,
+        handlers: ClickableHandlers,
+    ) {
+        const renderedChildren = this.props.children(eventState);
+
+        return React.cloneElement(renderedChildren, {
```

Comment:
Why do we clone the children?

Reply:
Oh, it's so we can add these extra attributes. Interesting.

Reply:
exactly! this is to make sure we pass the right props to the child component.